src/Microsoft.TemplateEngine.Cli/IInstaller.cs

```diff
@@ -10,5 +10,14 @@ public interface IInstaller : IInstallerBase
         void InstallPackages(IEnumerable<string> installationRequests, IList<string> nuGetSources, bool debugAllowDevInstall);
 
         void InstallPackages(IEnumerable<string> installationRequests, IList<string> nuGetSources, bool debugAllowDevInstall, bool interactive);
+
+        /// <param name="optionalWorkloadRequests">Requests from <paramref name="installationRequests"/> which are part of an optional workload.</param>
+        void InstallPackages(IEnumerable<string> installationRequests, IList<string> optionalWorkloadRequests, IList<string> nuGetSources);
```

Comment:
Wondering what you'd think about this:

```suggestion
        void InstallPackages(IEnumerable<InstallationRequest> installationRequests, IList<string> nuGetSources);
```

and

```C#
struct InstallationRequest
{
    public string FilePath; // Is it a path? It's not clear from the current interface what the string means.
    public bool IsOptionalWorkload;
}
```

Compared to the current proposal it is:
- More future proof - add more fields to `InstallationRequest` in the future, instead of adding overloads with even more parameters.
- More readable / self-explanatory - since it explicitly names the string.